ts/modules/full-screen.src.ts

```diff
@@ -18,63 +15,194 @@ import H from '../parts/Globals.js';
-     * @param {Highcharts.HTMLDOMElement} container
-     *        Chart container's parent
-     * @return {void}
-     */
-    init: function (
-        this: Highcharts.FullScreen,
-        container: Highcharts.HTMLDOMElement
-    ): void {
-
-        let promise;
-
-        if (container.requestFullscreen) {
-            promise = container.requestFullscreen();
+
+// Helping function - to open fullscreen, use toggleFullscreen() instead.
+H.Chart.prototype.openFullscreen = function (): void {
+    const chart = this,
+        menuItems = chart.options.exporting?.buttons?.contextButton.menuItems;
+
+    // Handle closeFullscreen() method when user clicks 'Escape' button.
+    chart.unbindFullscreenEvent = H.addEvent(
+        chart.container.ownerDocument, // chart's document
+        chart.browserProps.fullscreenChange,
+        function (): void {
+            // Handle lack of async of browser's fullScreenChange event.
+            if (chart.isFullscreen) {
+                chart.isFullscreen = false;
+                chart.closeFullscreen();
+            } else {
+                chart.isFullscreen = true;
+            }
+        }
+    );
+
+    if (chart.container.parentNode instanceof Element) {
+        chart.container.parentNode[chart.browserProps.requestFullscreen]();
+    }
+
+    // Replace button text. When chart.toggleFullscreen() will be fired customly
+    // by user before exporting context button is created, text will not be
+    // replaced - it's on the user side.
+    if (
+        chart.exportDivElements &&
+        chart.exportDivElements.length &&
+        menuItems &&
+        chart.options.lang &&
+        chart.options.lang.exitFullscreen
+    ) {
+        chart.exportDivElements[menuItems.indexOf('viewFullscreen')].innerHTML =
+        chart.options.lang.exitFullscreen;
+    }
+
+    H.addEvent(chart, 'destroy', chart.unbindFullscreenEvent);
+};
+
+// Helping function - to close fullscreen, use toggleFullscreen() instead.
+H.Chart.prototype.closeFullscreen = function (): void {
+    const chart = this,
+        exportingOptions = chart.options.exporting,
+        exportDivElements = chart.exportDivElements,
+        menuItems = exportingOptions?.buttons?.contextButton.menuItems;
+
+    // Don't fire exitFullscreen() when user exited using 'Escape' button.
+    if (
+        chart.isFullscreen &&
+        chart.container.ownerDocument instanceof Document
+    ) {
+        chart.container.ownerDocument[chart.browserProps.exitFullscreen]();
+    }
+
+    // Replace button text.
+    if (
+        exportDivElements &&
+        exportDivElements.length &&
+        exportingOptions?.menuItemDefinitions?.viewFullscreen &&
+        menuItems &&
+        chart.options.lang?.viewFullscreen
+    ) {
+        exportDivElements[menuItems.indexOf('viewFullscreen')].innerHTML = (
+            exportingOptions.menuItemDefinitions.viewFullscreen.text ||
+            chart.options.lang.viewFullscreen
+        );
+    }
```

Comment:
This code block is very similar to the above - refactor it out to a function. It can be local to the module.